addons/studio-invite-button/userscript.js

```diff
@@ -0,0 +1,64 @@
+export default async function ({ addon, global, console }) {
+  // RegExp to test for a valid url and get the studio id
+  const urlRegex = /^https?:\/\/scratch\.mit\.edu\/studios\/(\d+)\/comments($|\/)/;
+  // invites a user to the studio
+  function inviteToStudio(user) {
+    return fetch(
+      `https://scratch.mit.edu/site-api/users/curators-in/${
+        urlRegex.exec(window.location.href)[1]
+      }/invite_curator/?usernames=${user}`,
+      {
+        headers: {
+          "x-csrftoken": addon.auth.csrfToken,
+          "x-requested-with": "XMLHttpRequest",
+          referer: "https://scratch.mit.edu",
+        },
+        body: null,
+        method: "PUT",
+      }
+    );
+  }
+  addon.tab.waitForElement("#description").then((description) => {
+    // checks if the user is a manager and the URL is correct
+    if (description.classList.contains("editable") && urlRegex.test(window.location.href)) {
+      Array.from(document.getElementsByClassName("comment")).forEach((comment) => {
```

Comment:
This won't work for newly created comments. Do a `addon.tab.waitForElement` for `.comment`.

Reply:
Even though, the user should be seeing it on all comments.

Reply:
Okay, but how would addon.tab.waitForElement handle new comments?

Reply:
put it in a forever loop and use an `await`... smth like
```js
while (true) {
const comment = await addon.tab.waitForElement(".comment", { markAsSeen: true });
...
}
```

Reply:
do I have to wrap it in an async IIFE?

Reply:
> do I have to wrap it in an async IIFE?

No, everything is already inside an async function.

Reply:
you need to convert
![image](https://user-images.githubusercontent.com/61329810/114917631-cac95680-9deb-11eb-8db3-18eab9c52494.png)
to async tho

Reply:
You should actually remove that...

Reply:
@RedGuy12 told me to use addon.tab.waitForElement in that case, are you saying I should not?

Reply:
i said for the comment not for the desc

Reply:
If it's needed in order to check if you are in a studio, then you can keep it and make it async... and also remove the regex.

Reply:
What's wrong with the regex? I need to test if it's in the studio's comments, and find the studio ID.

Reply:
> What's wrong with the regex? I need to test if it's in the studio's comments, and find the studio ID.

You can change the match pattern for the script to `https://scratch.mit.edu/studios/*/comments/` and get the ID as `location.pathname.split("/")[2]`.

Reply:
Thanks, someone in the discord server told me to just test if the user was on /comments from inside the code